reconcile/aws_version_sync/integration.py

```diff
@@ -170,8 +176,34 @@ def get_aws_metrics(
                         resource_engine_version=m["engine_version"],
                     )
                     for m in prom_get_func(
-                        cluster.prometheus_url,
-                        {"query": "aws_resources_exporter_rds_engineversion"},
+                        url=cluster.prometheus_url,
+                        params={"query": "aws_resources_exporter_rds_engineversion"},
+                        token=token,
+                        timeout=timeout,
+                    )
+                ]
+                # ElastiCache resources
+                metrics += [
+                    ExternalResource(
+                        provider="aws",
+                        provisioner=ExternalResourceProvisioner(
+                            uid=m["aws_account_id"]
+                        ),
+                        resource_provider="elasticache",
+                        # cache_name is the ElastiCache node name, which derived from replication_group_id not resource_identifier!
+                        resource_identifier=elasticache_replication_group_id_to_identifier.get(
+                            node_name_to_cache_name(m["cache_name"]),
+                            node_name_to_cache_name(m["cache_name"]),
+                        ),
```

Comment:
Good point, thank you, Di. @ninabauer will fix that in the `aws-resource-exporter` https://issues.redhat.com/browse/APPSRE-8586?focusedId=23740171&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23740171

Reply:
That means we can directly use `ReplicationGroupId` as resource identifier, no need to map again, right?

Reply:
The `ReplicationGroupId` may differ from the app-interface resource identifier; we still need the mapping. the `resource identifier` is used to find the correct external resource in our namespace files.

Reply:
I only find one usage [here](https://github.com/app-sre/qontract-reconcile/blob/d76e99ae64bbb0a13ad93870ac4d8354384e1ed3/reconcile/utils/terrascript_aws_client.py#L2183), it could be override in somewhere else. Anyway make another mapping should be safe.